src/xo-app/self/dashboard/index.js

```diff
@@ -0,0 +1,317 @@
+      } else {
+        resolvedObjects[type].push(object)
+      }
+    })
+
+    return resolvedSet
+  })
+)
+
+// ===================================================================
+
+@propTypes({
+  entities: propTypes.array.isRequired
+})
+export class Entities extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      groups: [],
+      users: []
+    }
+  }
+
+  componentWillMount () {
+    const unsubscribeGroups = subscribeGroups(groups => {
+      this.setState({
+        groups: keyBy(groups, 'id')
+      })
+    })
+    const unsubscribeUsers = subscribeUsers(users => {
+      this.setState({
+        users: keyBy(users, 'id')
+      })
+    })
+
+    this.componentWillUnmount = () => {
+      unsubscribeGroups()
+      unsubscribeUsers()
+    }
+  }
+
+  render () {
+    const { state } = this
+
+    return (
+      <div>
+        {map(this.props.entities, id => {
+          const entity = state.users[id] || state.groups[id]
+          if (!entity) {
+            return <span key={id} className='m-r-1'>{_('unknownResourceSetValue')}</span>
+          }
+
+          if (entity.email) {
+            return (
+              <span key={id} className='m-r-1'>
+                <Icon icon='user' /> {entity.email}
+              </span>
+            )
+          }
+
+          return (
+            <span key={id} className='m-r-1'>
+              <Icon icon='group' /> {entity.name}
+            </span>
+          )
+        })}
+      </div>
+    )
+  }
+}
+
+// ===================================================================
+
+const OBJECT_TYPE_TO_ICON = {
+  'VM-template': 'vm',
+  'network': 'network',
+  'SR': 'sr'
+}
+
+export const ObjectP = propTypes({
+  object: propTypes.object.isRequired
+})(connectStore(() => {
+  const getPool = createGetObject(
+    (_, props) => props.object.$pool
+  )
+
+  return (state, props) => ({
+    pool: getPool(state, props)
+  })
+})(({ object, pool }) => {
+  const icon = OBJECT_TYPE_TO_ICON[object.type]
+  const { id } = object
+
+  return (
+    <span key={id} className='m-r-1'>
+      <Icon icon={icon} /> {object.name_label || id} {pool && `(${pool.name_label || pool.id})`}
+    </span>
+  )
+}))
+
+// ===================================================================
+
+class ResourceSet extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {}
+  }
+
+  _updateOpen = () => {
+    this.setState({
+      open: !this.state.open
+    })
+  }
+
+  render () {
+    const { open } = this.state
+    const { resourceSet } = this.props
+    const {
+      limits: {
+        cpus,
+        disk,
+        memory
+      } = {}
+    } = resourceSet
+
+    return (
+      <div className='p-b-1'>
+        <button className='card-header btn btn-lg btn-block' onClick={this._updateOpen}>
+          {resourceSet.name} <Icon icon={`chevron-${open ? 'up' : 'down'}`} />
+        </button>
+        <Panel className='p-t-1' collapsible expanded={open}>
```

Comment:
`Panel` is a Bootstrap 3 component, not 4, you should create your own card component or update React Bootstrap.